proposals/csharp-10.0/pattern-match-span-of-char-on-string.md

```diff
@@ -0,0 +1,49 @@
+<!--
+New language feature proposals should fully fill out this template. This should include a complete detailed design, which describes the syntax of the feature, what that syntax means, and how it affects current parts of the spec. Please make sure to point out specific spec sections that need to be updated for this feature. If you do not have the knowledge or experience to fill this out completely, that's perfectly alright: please open this proposal as a Discussion instead (https://github.com/dotnet/csharplang/discussions/new) and the community can generally discuss the proposal in less formal terms.
+-->
+# Pattern match `Span<char>` on a constant string
+
+* [x] Proposed
+* [x] Prototype: Completed
+* [x] Implementation: In Progress
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Permit pattern matching a `Span<char>` and a `ReadOnlySpan<char>` on a constant string.
+
+## Motivation
+[motivation]: #motivation
+
+For perfomance, usage of `Span<char>` and `ReadOnlySpan<char>` is preferred over string in many scenarios. The framework has added many new APIs to allow you to use `ReadOnlySpan<char>` in place of a `string`.
+
+A common operation on strings is to use a switch to test if it is a particular value, and the compiler optimizes such a switch. However there is currently no way to do the same on a `ReadOnlySpan<char>` efficiently, other than implementing the switch and the optimization manually.
+
+In order to encourage adoption of `ReadOnlySpan<char>` we allow pattern matching a `ReadOnlySpan<char>`, on a constant `string`, and optimize such switches on `ReadOnlySpan<char>` in the same way as switches on `string`.
+
+## Detailed design
+[design]: #detailed-design
+
+We alter the [spec](../csharp-7.0/pattern-matching.md#constant-pattern) for constant patterns as follows (the proposed addition is shown in bold):
+
+> A constant pattern tests the value of an expression against a constant value. The constant may be any constant expression, such as a literal, the name of a declared `const` variable, or an enumeration constant, or a `typeof` expression.
```

Comment:
I would either end with `, etc ...` to make it clear there are missing entries here or take the time to fill them all out. One example I see missing is `nameof()`.

Reply:
Done